### PR TITLE
[ONNX] Delete ONNXProgramSerializer

### LINKED ISSUE #135182: [ONNX] Remove ONNXProgramSerializer
Remove ONNXProgramSerializer. Serialization is handled by the IR.

### PR DESCRIPTION
**Pull request by titaiwangms**
## Context

Fixes #135182 

## BC Breaking

With the introduction of [ONNX IR](https://github.com/microsoft/onnxscript/blob/main/onnxscript/ir/README.md), ONNXProgram.model (ir.Model) is now serizalized instead.

textproto, onnxtext, and json format are supported by default when calling ONNXProgram.save() with a correspoinding file extension.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/_internal/_exporter_legacy.py [hunk 1]
```diff
@@ -1031,10 +924,24 @@ def save(
...
+                            destination,
+                            save_as_external_data=True,
+                            all_tensors_to_one_file=True,
+                        )
             else:
                 try:
-                    serializer.serialize(self, destination)
+                    if not isinstance(self.model_proto, onnx.ModelProto):  # type: ignore[attr-defined]
+                        raise ValueError(
+                            "onnx_program.ModelProto is not an onnx.ModelProto"
```
Comment (justinchuby) [inline torch/onnx/_internal/_exporter_legacy.py:942]: I don't think this check is required?

File: torch/onnx/_internal/_exporter_legacy.py [hunk 2]
```diff
@@ -1031,10 +924,24 @@ def save(
...
+                        onnx.save_model(  # type: ignore[attr-defined]
+                            self.model_proto,
+                            destination,
+                            save_as_external_data=True,
+                            all_tensors_to_one_file=True,
+                        )
             else:
                 try:
-                    serializer.serialize(self, destination)
+                    if not isinstance(self.model_proto, onnx.ModelProto):  # type: ignore[attr-defined]
```
Comment (justinchuby) [inline torch/onnx/_internal/_exporter_legacy.py:940]: I don't think this else branch is needed?

Comment (justinchuby) [reply torch/onnx/_internal/_exporter_legacy.py:940]: I see destination may not be str. NVM

File: torch/onnx/_internal/_exporter_legacy.py (hunk 1 above)
Comment (titaiwangms) [reply torch/onnx/_internal/_exporter_legacy.py:942]: I will follow up on this

**Pull request closed without merging**